x/bank/keeper/keeper.go

```diff
@@ -131,6 +139,13 @@ func (k BaseKeeper) DelegateCoins(ctx context.Context, delegatorAddr, moduleAccA
 		return errorsmod.Wrap(sdkerrors.ErrInvalidCoins, amt.String())
 	}
 
+	err := k.BlockBeforeSend(ctx, delegatorAddr, moduleAccAddr, amt)
+	if err != nil {
+		return err
+	}
+	// call the TrackBeforeSend hooks and the BlockBeforeSend hooks
+	k.TrackBeforeSend(ctx, delegatorAddr, moduleAccAddr, amt)
+
```

Comment:
// call the TrackBeforeSend hooks and the BlockBeforeSend hooks
	err := k.BlockBeforeSend(ctx, delegatorAddr, moduleAccAddr, amt)
	if err != nil {
		return err
	}
	k.TrackBeforeSend(ctx, delegatorAddr, moduleAccAddr, amt)

Reply:
The comment should be located before calling the BlockBeforeSend function

Reply:
How are `BlockBeforeSend` and `TrackBeforeSend` functions will be used in the implementation of the the *Composite* module?

Reply:
Fixed comment, these hooks are currently unused but we might need them later on such like Osmosis has for the token factory, so leaving them there since they cause no harm.